runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/WindmillTimeUtilsTest.java

```diff
@@ -56,6 +56,7 @@ public void testWindmillToHarnessTimestamp() {
     assertEquals(new Instant(-17), windmillToHarnessTimestamp(-16987));
     assertEquals(new Instant(-17), windmillToHarnessTimestamp(-17000));
     assertEquals(new Instant(-18), windmillToHarnessTimestamp(-17001));
+    assertEquals(BoundedWindow.TIMESTAMP_MIN_VALUE, windmillToHarnessTimestamp(Long.MIN_VALUE + 1));
```

Comment:
Did you verify this test fails without the change to WindmillTimeUtils?
Looking again I'm wondering if the divideAndRoundDown would have already ended up with this value.